torch/optim/sgd.py

```diff
@@ -50,6 +50,13 @@ class SGD(Optimizer):
 
     def __init__(self, params, lr=required, momentum=0, dampening=0,
                  weight_decay=0, nesterov=False):
+        if not 0.0 <= lr:
+            raise ValueError("Invalid learning rate: {}".format(lr))
+        if not 0.0 <= momentum <= 1.0:
+            raise ValueError("Invalid momentum value: {}".format(momentum))
```

Comment:
This is questionable. In practice only a momentum value less than 1 is used, but should not restrict the user from having higher momentum.

Reply:
Agreed. Will change this. Can you also review other ranges?